packages/connect/src/connector/extension.ts

```diff
@@ -57,95 +50,204 @@ export const createScClient = (): ScClient => {
     jsonRpcCallback?: JsonRpcCallback,
     potentialRelayChainIds = [] as string[],
   ): Promise<Chain> => {
-    const chainId = getRandomChainId()
-    const postToExtension = (msg: HeaderlessToExtension) => {
-      window.postMessage(
-        {
-          ...msg,
-          origin: "substrate-connect-client",
-          chainId,
-        },
-        "*",
-      )
-    }
+    type ChainState =
+      | {
+          state: "wait-ack-or-err"
```

Comment:
should we name this state: "pending"? I mean - it does not really matter - just sounds better than "wait-ack-or-err"